neo/Storage/Implementation/AbstractDBImplementation.py

```diff
@@ -0,0 +1,56 @@
+from abc import ABC, abstractmethod
+"""
+Description:
+    Alternative backends can be added by implementing the required methods
+    of this abstract class.
+Usage:
+    The dynamically generated class coming from the database factory inherits
+    from the abstract class, means the generated class cannot be used if not
+    all methods defined in this class are overwritten.
+
+    If this class is extended, make sure you extend also all other database
+    implementations.
+
+    For a more detailed information on the methods and how to implement a new
+    database backend check out:
+    neo.Storage.Implementation.LevelDB.LevelDBClassMethods
+"""
+
+
+class AbstractDBImplementation(ABC):
+
```

Comment:
we need to add a mandatory `def __init__(self, path):` that needs to be overridden.

Reply:
Done.